lib/utils/promote.ts

```diff
@@ -206,28 +221,12 @@ async function getOrSelectApplication(
 		.value();
 
 	if (!appName) {
-		const options = {
-			$filter: { device_type: { $in: compatibleDeviceTypes } },
-		};
-
-		// No application specified, show a list to select one.
-		const applications = await sdk.models.application.getAll(options);
-
-		if (applications.length === 0) {
-			const shouldCreateApp = await form.ask({
-				message:
-					'You have no applications this device can join.\n' +
-					'Would you like to create one now?',
-				type: 'confirm',
-				default: true,
-			});
-			if (shouldCreateApp) {
-				return createApplication(sdk, deviceType);
-			}
-			process.exit(1);
-		}
-
-		return selectAppFromList(applications);
+		return createOrSelectAppOrExit(
```

Comment:
I'd suggest any function that has more than three params (and I actually benchmark two personally), should have a defined interface and should be passed as a params object.

Reply:
I'm not even sure this function should exist -- some refactoring is probably in order, but not under this PR. I've added the following comments a few lines below:
```
// TODO: revisit this function's purpose. It was refactored out of
// `getOrSelectApplication` above in order to satisfy some resin-lint v3
// rules, but it looks like there's a fair amount of duplicate logic.
```
The reasons for me to move the block of code to a separate function were a few occurrences of "shadowed name" linter warnings, plus the fact that the original function was already quite large.

Reply:
OK, cool.